### PR TITLE
refresh LightningModule API docs

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

This PR includes several improvements to the API docs in `lightning.py`. 

- Added missing docs to public methods and properties (the lack thereof was brought to our attention by @ananthsub)
- Fixed typos
- Fixed whitespace leading to malformatted html
- Fixed linking between methods and attributes to improve navigation (where possible and appropriate)

The build of the html can be viewed [here](https://119173-178626720-gh.circle-artifacts.com/0/html/api/pytorch_lightning.core.lightning.html) or by going to the workflows `ci-tests -> click on tab "artifacts" -> navigate to html/api/pytorch_lightning.core.lightning.html`

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/core/lightning.py [hunk 1]
```diff
@@ -1662,6 +1688,21 @@ def tbptt_split_batch(self, batch, split_size):
         return splits
 
     def summarize(self, mode: Optional[str] = "top", max_depth: Optional[int] = None) -> Optional[ModelSummary]:
+        """
+        Summarize this LightingModule.
```
Comment (awaelchli) [inline pytorch_lightning/core/lightning.py:1692]:
```suggestion
        Summarize this LightningModule.
```

File: pytorch_lightning/core/lightning.py [hunk 2]
```diff
@@ -117,6 +117,17 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
         self._metric_attributes: Optional[Dict[int, str]] = None
 
     def optimizers(self, use_pl_optimizer: bool = True) -> Union[Optimizer, List[Optimizer], List[LightningOptimizer]]:
+        """
+        Returns the optimizer(s) that are being used during training. Useful for manual optimization.
+
+        Args:
+            use_pl_optimizer: If ``True``, will wrap the optimizer(s) in a
+                :class:`~pytorch_lighting.core.optimizer.LightningOptimizer` for automatic handling of precision and
```
Comment (awaelchli) [inline pytorch_lightning/core/lightning.py:125]:
```suggestion
                :class:`~pytorch_lightning.core.optimizer.LightningOptimizer` for automatic handling of precision and
```

**Review by SeanNaren — APPROVED** [review]
Thanks @awaelchli :)

**Review by kaushikb11 — APPROVED** [review]
Great!

**Review by ethanwharris — APPROVED** [review]

**Review by tchaton — APPROVED** [review]

**Pull request merged**